platforms/core-configuration/model-core/src/main/java/org/gradle/internal/metaobject/BeanDynamicObject.java

```diff
@@ -275,9 +276,13 @@ public DynamicInvokeResult getProperty(String name) {
                 // Do not check for opaque properties when implementing PropertyMixIn, as this is expensive
             }
 
-            // TODO: Deprecate and nag about this
             MetaMethod metaMethod = lookupMethod(metaClass, "is" + StringUtils.capitalize(name), MetaClassHelper.EMPTY_CLASS_ARRAY);
             if (metaMethod != null && metaMethod.getReturnType().equals(Boolean.class)) {
+                DeprecationLogger.deprecateAction("Referencing '" + name + "' as a property declared with an 'is-' method with a Boolean type on " + getDisplayName())
```

Comment:
This wording is a little weird to me. WDYT:
```suggestion
                DeprecationLogger.deprecateAction("Referencing property '" + name + "' that was declared with an 'is-' method with a Boolean type on " + getDisplayName())
```